### ISSUE TITLE
Question: how to hook in to subcommand Help...

### ISSUE THREAD

**Issue by JRAndreassen** [iqlusioninc/abscissa#354]
Hi..

I'm trying to get help implemented for an option...
```
target/release/my_agent.exe help scan
my_agent 0.1.1


USAGE:
    my_agent scan <OPTIONS>

DESCRIPTION:
    Perform Data Scan

POSITIONAL ARGUMENTS:
    extra

FLAGS:
    -d, --devicetype DEVICE-TYPE 
    -U, --url URL             URL or ip of target
    -m, --method METHOD       Authentication Method
    -u, --username USERNAME   Authentication User Name (root|admin|apiuser)
    -p, --password PASSWORD   Authentication Password
    --token TOKEN             Authentication token
    -c, --clientid CLIENTID   Authentication ClientId
    --clientsecret CLIENTSECRET
    --tenantid TENANTID       Authentication Tenant ID (M$)
    -e, --emailuser EMAILUSER E-Mail user name
    -r, --realm REALM         Authentication Realm
    -A, --action ACTION       SubDevice Instance or Action verb (reset|on|off...)
    -P, --actionparam ACTION-PARAM
    -C, --credentialname my-CREDS
    -v, --verbose             Show verbose message
    -t, --timeout TIMEOUT-MS  Connect timeout in milliseconds (Default 20sec/21000ms (default: 21000)
    -T, --timing              Activate timing mode (default: false)
    -l, --log_message LOG-MESSAGE
    -L, --log_path LOG-PATH   Log file base path
```
the option : "DEVICE-TYPE"
has a dynamic list of options...
Is there a way to hook into the help system to generate a list ?

Also... Is there a way to hook in with the "Graceful" shutdown ?
Given that my app uses Tokio:

```rust
    fn run(&self) {
        let _ = abscissa_tokio::run(&APPLICATION, 
            async{
...
                let upload_interval = 
                    Duration::from_secs(self.upload_interval.unwrap_or(60));
                info!("Starting agent Loop");
                loop {
                    APPLICATION.read().get_handler_controller().read()
                        .handle_tasks_until(&mut todo_list, upload_interval).await;
                    self.dump_data(&mut todo_list);
                    // &&##&& Termination condition ???
                }
            }
        );
    }
```
The application ahs the Shutdown method...
It would be handy if the "Runnable had a "shutdown(&mut self, shutdown: Shutdown)"
Or did I miss something ?

Thanks again for sharing this project...
It's been a great time saver...
  JR

**Comment by tony-iqlusion** [comment]
> Is there a way to hook into the help system to generate a list ?

Not really. For what it's worth, we're looking at replacing the library we use for command line option parsing: #298

**Comment by tony-iqlusion** [comment]
Closed by #562 (clap v3 migration)
